cmd/dep/import_analyzer.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"github.com/golang/dep"
+	"github.com/golang/dep/internal/gps"
+)
+
+// importer handles importing configuration from other dependency managers into our format
+type importer interface {
+	Import(path string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error)
+	HasConfig(dir string) bool
+}
+
+// importAnalyzer imports existing dependency management configuration,
+// from both dep and external tools.
+type importAnalyzer struct {
+	loggers *dep.Loggers
+	sm      gps.SourceManager
+}
+
+func newImportAnalyzer(loggers *dep.Loggers, sm gps.SourceManager) importAnalyzer {
+	return importAnalyzer{loggers: loggers, sm: sm}
+}
+
+func (a importAnalyzer) importManifestAndLock(dir string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	importers := []importer{
+		newGlideImporter(a.loggers, a.sm),
+	}
+
+	for _, i := range importers {
+		if i.HasConfig(dir) {
+			if a.loggers.Verbose {
+				a.loggers.Err.Printf("Importing %T configuration for %s. Run with -skip-tools to skip.", i, pr)
+			}
+			return i.Import(dir, pr)
+		}
+	}
+
+	return nil, nil, nil
+}
+
+func (a importAnalyzer) DeriveRootManifestAndLock(dir string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	return a.importManifestAndLock(dir, pr)
+}
+
+func (a importAnalyzer) DeriveManifestAndLock(dir string, pr gps.ProjectRoot) (gps.Manifest, gps.Lock, error) {
+	// Ignore other tools if we find dep configuration
+	var depAnalyzer dep.Analyzer
+	if depAnalyzer.HasConfig(dir) {
+		return depAnalyzer.DeriveManifestAndLock(dir, pr)
+	}
+
+	// The assignment back to an interface prevents interface-based nil checks from failing later
```

Comment:
Yeah I struggled to figure out how to reuse that logic without causing typed nil troubles. Let me know if it's preferable to copy/paste code vs. calling the function and coercing the nils.